CMSgov/design-system
# [WNMGDS-877] Update type sizing classes and variables

## Summary
Added new agnostic variables and classes to replace the semantic variables and classes in the future. 
This PR adds new classes and removes the old from documentation. All old classes will still work and are still part of the codebase to prevent breaking changes. 

**Example of semantic vs agnostic**
`$h1-font-size` is a semantic variable because it refers to a specific element, the `<h1>` heading. 
`$font-size-xl` is an agnostic variable because it does not refer to a specific element.
`.ds-u-font-size--h1` is a semantic class because it refers to a specific element, the `<h1>` heading. 
`.ds-u-font-size--xl` is an agnostic class because it does not refer to a specific element.

Switching to agnostic naming is less confusing because you don't end up with code that uses a class that refers to a different element. For example, in the code below we have an `<h1>` element that uses a class name that refers to `<h3>`
```
<h1 class="ds-h3">Heading one that is styled with H3</h1>
```

### Added
- Added agnostic font size variables 
    Using current variables for values to ensure nothing breaks 
```
$font-size-5xl: $display-font-size !default; //60px
$font-size-4xl: $title-font-size !default; // 48px
$font-size-3xl: $h1-font-size !default; // 36px - h1
$font-size-2xl: $h2-font-size !default; // 24px - h2
$font-size-xl: $h3-font-size !default; // 21px - h3
$font-size-lg: $lead-font-size !default; // 18px - h4 - lead - body large
$font-size-md: $base-font-size !default; // 16px - h5 - base - body medium
$font-size-sm: $small-font-size !default; // 14px - h6 - body small
```
- Added agnostic font-size utility class names
   
```
.ds-u-font-size--[display|title|h1|h2|h3|h4|lead|base|small]
```

### Changed
- Updated documentation pages for typography and font-sizing 
- Updated type examples to use new agnostic class names
- Updated variables in components to use agnostic variable names for font sizing

### Deprecated
With this PR H tag variable and class names are now marked for deprecation in the future in favor of the new agnostic naming convention. 

## How to test
Check out the Demo URL pages 
- [Typography](http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-877-update-font-sizing-classes/styles/typography/)
- [Font size utility](http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-877-update-font-sizing-classes/utilities/font-size/)

- Test this branch on both the Mgov and HCgov child design systems by updating the package.json to point to this branch. There should be no visible issue with font sizing.
